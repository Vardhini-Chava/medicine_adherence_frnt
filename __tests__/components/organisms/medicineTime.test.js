import Enzyme, { shallow, render } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { Provider } from "react-redux";
import React from "react";
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json";
import Box from "../../../src/components/organisms/medicineTime";
import enableHooks from "jest-react-hooks-shallow";
enableHooks(jest);

Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureStore([]);

const findNodeByTestId = (wrapper, testID) => {
  return wrapper.findWhere((node) => {
    return node.prop("testID") === testID;
  });
};
describe("test collector category", () => {
  let store;
  store = mockStore({
    field: {
      name: "hello",
      value: "San",
    },
  });
  it("test category", () => {
    const wrapper = render(
      <Provider store={store}>
        <Box />
      </Provider>
    );

    expect.assertions(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("test open save button", () => {
    const mockFn = jest.fn();
    const wrapper = shallow(<Box updateMed={mockFn} />);
    wrapper.find("BouncyCheckbox#date");
  });

  it("not open save button", () => {
    const mockFn = jest.fn();
    let updatetimes = jest.fn();
    const props = {
      time: "BouncyCheckbox",
      updatetimes: jest.fn(),
    };
    const wrapper = shallow(<Box props={props} updatetimes={updatetimes} />);
    const tree = findNodeByTestId(wrapper, "BouncyCheckbox");
    expect(wrapper).toBeDefined();
    tree.props().onPress();
  });
});
