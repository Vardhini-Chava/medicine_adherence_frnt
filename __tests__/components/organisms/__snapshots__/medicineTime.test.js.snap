// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test collector category test category 1`] = `
<view
  style="padding:15px;padding-left:30px;margin-top:14px"
>
  <view
    accessibilitystate="[object Object]"
    focusable="true"
    style="align-items:center;flex-direction:row;opacity:1"
    testid="BouncyCheckbox"
  >
    <view
      style="transform:[object Object];width:22px;height:22px;border-width:1.3px;border-color:#3743ab;border-radius:11px;align-items:center;justify-content:center;background-color:#FFFFFF"
    />
    <view
      style="0:[object Object]"
    >
      <text
        style="0:[object Object];1:[object Object]"
      />
    </view>
  </view>
  <text
    style="color:red;margin-left:40px"
  >
    Not Taken
  </text>
</view>
`;
